src/System.Linq/src/System/Linq/Lookup.cs

```diff
@@ -182,7 +196,7 @@ internal List<TResult> ToList<TResult>(Func<TKey, IEnumerable<TElement>, TResult
             return _count;
         }
 
-        public IEnumerable<TResult> ApplyResultSelector<TResult>(Func<TKey, IEnumerable<TElement>, TResult> resultSelector)
+        public IEnumerator<TResult> ApplyResultSelector<TResult>(Func<TKey, IEnumerable<TElement>, TResult> resultSelector)
```

Comment:
Yep. It's either an extra generic type fed to the Jitter or AOT, or if we try to be clever to avoid that it's about 50 lines of code, all to avoid a single virtual method calls made before N × 2 + 1 method calls anyway and not likely to be in a hot path.

There's something satisfying about the idea of having an overloaded constructor depending on whether you want to use the iterator as an enumerable or an enumerator, but if I were to code it for that reason alone, you should probably reject the PR because of the above :smile: